src/main/java/de/rub/nds/scanner/core/probe/ScannerProbe.java

```diff
@@ -282,155 +100,49 @@ public final void putIfEqual(
         }
     }
 
-    public final void putIfEqual(
-            AnalyzedProperty determiningProperty,
-            AnalyzedProperty propertyToSet,
-            Map<?, ?> actualResult,
-            String notApplicableReason,
-            TestResult expectedValue) {
-        if (!propertiesMap.containsKey(determiningProperty)
-                || propertiesMap.get(determiningProperty) == null
-                || !propertiesMap.get(determiningProperty).equals(expectedValue)) {
-            put(propertyToSet, new NotApplicableResult(propertyToSet, notApplicableReason));
-        } else {
-            put(propertyToSet, actualResult);
+    private TestResult convertToResult(AnalyzedProperty property, Object result) {
+        // java 17 could use pattern matching :S
```

Comment:
change to TODO: change to pattern matching with Java 17

Reply:
Do you think we'll ever switch to java 17(+)? :grimacing:

Reply:
I don't think the changes will be super complicated. The only thing I remember that made the change from Java 8 to 11 tedious was the movement of dependencies